// golos-js/lib/auth/serializer/src/ChainTypes.js
"use strict";

var ChainTypes;

module.exports = ChainTypes = {};

ChainTypes.reserved_spaces = {
  relative_protocol_ids: 0,
  protocol_ids: 1,
  implementation_ids: 2
};

ChainTypes.operations = {
  // _16 for 0.16.4
  vote_16: 0,
  comment_16: 1,
  transfer_16: 2,
  transfer_to_vesting_16: 3,
  withdraw_vesting_16: 4,
  limit_order_create_16: 5,
  limit_order_cancel_16: 6,
  feed_publish_16: 7,
  convert_16: 8,
  account_create_16: 9,
  account_update_16: 10,
  witness_update_16: 11,
  account_witness_vote_16: 12,
  account_witness_proxy_16: 13,
  pow_16: 14,
  //
  custom: 15,
  //
  report_over_production_16: 16,
  delete_comment_16: 17,
  //
  custom_json: 18,
  //
  comment_options_16: 19,
  set_withdraw_vesting_route_16: 20,
  limit_order_create2_16: 21,
  challenge_authority_16: 22,
  prove_authority_16: 23,
  request_account_recovery_16: 24,
  recover_account_16: 25,
  change_recovery_account_16: 26,
  escrow_transfer_16: 27,
  escrow_dispute_16: 28,
  escrow_release_16: 29,
  pow2_16: 30,
  escrow_approve_16: 31,
  transfer_to_savings_16: 32,
  transfer_from_savings_16: 33,
  cancel_transfer_from_savings_16: 34,
  //
  custom_binary: 35,
  //
  decline_voting_rights_16: 36,
  reset_account_16: 37,
  set_reset_account_16: 38,
  comment_benefactor_reward_16: 39,
  //
  vote: 40,
  comment: 41,
  transfer: 42,
  transfer_to_vesting: 43,
  withdraw_vesting: 44,
  limit_order_create: 45,
  limit_order_cancel: 46,
  feed_publish: 47,
  convert: 48,
  account_create: 49,
  account_update: 50,
  witness_update: 51,
  account_witness_vote: 52,
  account_witness_proxy: 53,
  pow: 54,
  report_over_production: 55,
  delete_comment: 56,
  comment_options: 57,
  set_withdraw_vesting_route: 58,
  limit_order_create2: 59,
  challenge_authority: 60,
  prove_authority: 61,
  request_account_recovery: 62,
  recover_account: 63,
  change_recovery_account: 64,
  escrow_transfer: 65,
  escrow_dispute: 66,
  escrow_release: 67,
  pow2: 68,
  escrow_approve: 69,
  transfer_to_savings: 70,
  transfer_from_savings: 71,
  cancel_transfer_from_savings: 72,
  decline_voting_rights: 73,
  reset_account: 74,
  set_reset_account: 75,
  comment_benefactor_reward: 76,
  delegate_vesting_shares: 77,
  account_create_with_delegation: 78,
  comment_payout_extension: 79,
  asset_create: 80,
  asset_update: 81,
  asset_update_bitasset: 82,
  asset_update_feed_producers: 83,
  asset_issue: 84,
  asset_reserve: 85,
  asset_fund_fee_pool: 86,
  asset_settle: 87,
  asset_force_settle: 88,
  asset_global_settle: 89,
  asset_publish_feed: 90,
  asset_claim_fees: 91,
  call_order_update: 92,
  account_whitelist: 93,
  override_transfer: 94,
  proposal_create: 95,
  proposal_update: 96,
  proposal_delete: 97,
  bid_collateral: 98,
  // _16 for 0.16.4
  fill_convert_request_16: 99,
  author_reward_16: 100,
  curation_reward_16: 101,
  comment_reward_16: 102,
  liquidity_reward_16: 103,
  interest_16: 104,
  fill_vesting_withdraw_16: 105,
  fill_order_16: 106,
  shutdown_witness_16: 107,
  fill_transfer_from_savings_16: 108,
  hardfork_16: 109,
  comment_payout_update_16: 110,
  //
  fill_convert_request: 111,
  author_reward: 112,
  curation_reward: 113,
  comment_reward: 114,
  liquidity_reward: 115,
  interest: 116,
  fill_vesting_withdraw: 117,
  fill_order: 118,
  shutdown_witness: 119,
  fill_transfer_from_savings: 120,
  hardfork: 121,
  comment_payout_update: 122,
  return_vesting_delegation: 123,
  asset_settle_cancel: 124,
  fill_call_order: 125,
  fill_settlement_order: 126,
  execute_bid: 127
};

//types.hpp
ChainTypes.object_type = {
  "null": 0,
  base: 1
};
